tiled/_tests/test_asset_access.py

```diff
@@ -0,0 +1,69 @@
+import hashlib
+from pathlib import Path
+
+import pytest
+
+from ..catalog import in_memory
+from ..client import Context, from_context
+from ..client.utils import get_asset_filepaths
+from ..server.app import build_app
+from ..utils import path_from_uri
+
+
+@pytest.fixture
+def context(tmpdir):
+    catalog = in_memory(writable_storage=str(tmpdir))
+    app = build_app(catalog)
+    with Context.from_app(app) as context:
+        yield context
+
+
+@pytest.fixture
+def client(context):
+    client = from_context(context)
+    yield client
+
+
+def test_include_data_sources_method_on_self(client):
+    "Calling include_data_sources() fetches data sources on self."
+    client.write_array([1, 2, 3], key="x")
+    with pytest.warns(UserWarning):
+        # This fetches the sources with an additional implicit request.
+        client["x"].data_sources()
+    client["x"].include_data_sources().data_sources() is not None
```

Comment:
Am I missing some magics somewhere that's making them implicit or are you missing a bunch of `assert`s?